tests/lib/test_schema.py

```diff
@@ -0,0 +1,151 @@
+import yaml
+import sys
+import pprint
+
+def check_bool(value, expected):
+    if expected == 'false()' and value is False:
+        return 1
+    if expected == 'true()' and value is True:
+        return 1
+    print(value)
+    print(expected)
+    return 0
+
+def check_int(value, expected):
+    if (int(expected) == value):
+        return 1
+    print(value)
+    print(expected)
+    return 0
+
+def check_float(value, expected):
+    if expected == 'inf()':
+        if value == float("inf"):
+          return 1
+    elif expected == 'inf-neg()':
+        if value == float("-inf"):
+          return 1
+    elif expected == 'nan()':
+        if value != value or (value == 0.0 and value == 1.0):
+          return 1
+    elif (float(expected) == value):
+        return 1
+    else:
+        print(value)
+        print(expected)
+        return 0
+
+def check_str(value, expected):
+    if value == expected:
+        return 1
+    print(value)
+    print(expected)
+    return 0
+
+
+def _fail(input, test):
+    print("Input: >>" + input + "<<");
+    print(test);
+
+# The tests/data/yaml11.schema file is copied from
+# https://github.com/perlpunk/yaml-test-schema/blob/dev/data/schema-yaml11.yaml
+def test_implicit_resolver(data_filename, skip_filename, verbose=False):
+    types = {
+        'str':   [str,   check_str],
+        'int':   [int,   check_int],
+        'float': [float, check_float],
+        'inf':   [float, check_float],
+        'nan':   [float, check_float],
+        'bool':  [bool,  check_bool],
+    }
+    skipdata = yaml.load(open(skip_filename, 'rb'), Loader=yaml.SafeLoader)
+    skip_load = skipdata['load'];
+    skip_dump = skipdata['dump'];
+    if verbose:
+        print(skip_load)
+    tests = yaml.load(open(data_filename, 'rb'), Loader=yaml.SafeLoader)
+
+    i = 0;
+    fail = 0;
+    for input, test in sorted(tests.items()):
```

Comment:
If `i` is still needed:

```suggestion
    for i, (input, test) in enumerate(sorted(tests.items())):
```

Reply:
done